c2corg_api/views/document_changes.py

```diff
@@ -0,0 +1,160 @@
+            a `pagination_token`. This token is to be used to request the next
+            page.
+
+            `u=...` (optional)
+            Changes made by one user.
+
+            For more information about "continuation token pagination", see:
+            http://www.servicedenuages.fr/pagination-continuation-token (fr)
+        """
+        user_id = self.request.validated.get('u')
+        lang, token_id, token_time, limit = get_params(self.request)
+
+        if token_time is None:
+            token_time = datetime.datetime.now()
+
+        changes = get_changes_of_feed(token_id, token_time, limit, user_id)
+
+        doc_ids = []
+        for ch in changes:
+            doc_ids.append(ch.id)
+
+        return load_feed(doc_ids, limit, user_id)
+
+
+def get_changes_of_feed(token_id, token_time, limit, user_id=None):
+    query = DBSession. \
+        query(HistoryMetaData.id, HistoryMetaData.user_id,
+              HistoryMetaData.written_at). \
+        order_by(HistoryMetaData.written_at.desc())
+
+    # pagination filter
+    if token_id is not None and token_time:
+        query = query.filter(
+            or_(
+                HistoryMetaData.written_at < token_time,
+                and_(
+                    HistoryMetaData.written_at == token_time,
+                    HistoryMetaData.id > token_id)))
+
+    if user_id is not None:
+        query = query.filter(HistoryMetaData.user_id == user_id)
+
+    return query.limit(limit).all()
+
+
+def load_feed(doc_ids, limit, user_id=None):
+    if user_id:
+        doc_total = DBSession.query(HistoryMetaData.user_id)\
+            .filter(HistoryMetaData.user_id == user_id).count()
+    else:
+        doc_total = DBSession.query(HistoryMetaData.user_id).count()
+
+    doc_changes = DBSession.query(DocumentVersion) \
+        .options(load_only('history_metadata', 'lang'),
+                 joinedload('history_metadata').load_only(
+                    HistoryMetaData.id,
+                    HistoryMetaData.user_id,
+                    HistoryMetaData.comment,
+                    HistoryMetaData.written_at).
+                 joinedload('user').load_only(
+                    User.id,
+                    User.name,
+                    User.username,
+                    User.lang)) \
+        .options(load_only('document'),
+                 joinedload('document').load_only(
+                    Document.version,
+                    Document.document_id,
+                    Document.type,
+                    Document.quality)) \
+        .options(load_only('document_locales_archive'),
+                 joinedload('document_locales_archive').load_only(
+                    ArchiveDocumentLocale.title)) \
+        .order_by(desc(DocumentVersion.id)) \
+        .filter(DocumentVersion.history_metadata_id.in_(doc_ids)) \
+        .limit(limit) \
+        .all()
+
+    if not doc_changes:
+        return {'feed': []}
```

Comment:
Shouldn't you return ``total: 0`` as well? It would be more consistent with the case where the feed is actually not empty.